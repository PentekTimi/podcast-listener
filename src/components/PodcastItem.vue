<template>
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
  >
    <div>
      <router-link
        :to="{ name: 'podcast', params: { id: podcast.docID } }"
        class="font-bold block text-gray-600"
        >{{ podcast.modified_name }}</router-link
      >
      <span class="text-gray-500 text-sm">{{ podcast.display_name }}</span>
    </div>

    <div class="text-gray-600 text-lg">
      <span class="comments">
        <i class="fa fa-comments text-gray-600"></i>
        {{ podcast.comment_count }}
      </span>
    </div>
  </li>
</template>

<script>
export default {
  name: 'PodcastItem',
  props: ['podcast']
}
</script>
