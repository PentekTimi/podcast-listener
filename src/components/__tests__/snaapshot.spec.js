import PodcastItem from '@/components/PodcastItem.vue'
import { mount, RouterLinkStub } from '@vue/test-utils'

describe('snapshots PodcastItem.vue', () => {
  test('renders correctly', () => {
    const podcast = {
      docID: 'abc',
      modified_name: 'test',
      display_name: 'test',
      comment_count: 5
    }

    const wrapper = mount(PodcastItem, {
      props: { podcast },
      global: {
        components: {
          'router-link': RouterLinkStub
        }
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
