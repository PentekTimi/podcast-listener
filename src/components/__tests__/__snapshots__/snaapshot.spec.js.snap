// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshots PodcastItem.vue > renders correctly 1`] = `
<a>
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
  >
    <div
      class="max-w-[70%]"
    >
      <p
        class="font-bold block text-zinc-400 break-words text-wrap"
      >
        test
      </p>
      <span
        class="text-zinc-500 text-sm"
      >
        test
      </span>
    </div>
    <div
      class="text-zinc-600 text-lg"
    >
      
      <span
        class="comments"
      >
        <font-awesome-icon
          class="text-zinc-600"
          icon="fa fa-comment "
        />
         5
      </span>
      
    </div>
  </li>
</a>
`;
